### PR TITLE
Added view link parameter

### PR DESCRIPTION
**Pull request by janakkaneriya007**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #ISSUE_NUMBER .

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-moderation/classes/class-bp-moderation-activity-comment.php [hunk 1]
```diff
@@ -72,13 +72,39 @@ public static function get_content_owner_id( $activity_id ) {
 	 * @since BuddyBoss 2.0.0
 	 *
 	 * @param int $activity_comment_id activity id.
+	 * @param bool $view_link   add view link
 	 *
 	 * @return string
 	 */
-	public static function get_content_excerpt( $activity_comment_id ) {
+	public static function get_content_excerpt( $activity_comment_id, $view_link ) {
```
Comment (dipskakadiya) [inline src/bp-moderation/classes/class-bp-moderation-activity-comment.php:79]: @janakkaneriya007 `view_link ` default value should be `false` in all content type class

File: src/bp-templates/bp-nouveau/buddypress/moderation/moderation-reported-content-entry.php [hunk 2]
```diff
@@ -39,7 +39,7 @@
     </td>
     <td class="moderation-content-excerpt">
 		<?php
-		$content_excerpt = bp_moderation_get_content_excerpt( bp_get_moderation_item_id(), bp_get_moderation_item_type() );
+		$content_excerpt = bp_moderation_get_content_excerpt( bp_get_moderation_item_id(), bp_get_moderation_item_type(), false );
```
Comment (dipskakadiya) [inline src/bp-templates/bp-nouveau/buddypress/moderation/moderation-reported-content-entry.php:42]: @janakkaneriya007 As we adding default value for 3rd argument as `false` so no need to pass `false`

File: src/bp-moderation/screens/content-report-form.php [hunk 3]
```diff
@@ -53,16 +53,14 @@
...
-                <input type="button" class="bb-cancel-report-content"
-                       value="<?php esc_attr_e( 'Cancel', 'buddyboss' ); ?>"/>
+				</label>
+			</div>
+			<div class="form-item bp-hide">
+				<label for="report-note">
+					<textarea id="report-note" type="text" name="note" class="bp-other-report-cat"></textarea> </label>
+			</div>
+			<div class="form-item">
+				<input type="button" class="bb-cancel-report-content" value="<?php esc_attr_e( 'Cancel', 'buddyboss' ); ?>"/>
```
Comment (dipskakadiya) [inline src/bp-moderation/screens/content-report-form.php:63]: @janakkaneriya007 Spacing issue

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-activity/bp-activity-functions.php [hunk 4]
```diff
@@ -3508,6 +3508,16 @@ function bp_activity_delete_children( $activity_id, $comment_id ) {
...
+
+			/**
+			 * Fires after deleting all the chidren comment of comment.
+			 *
+			 * @since BuddyPress 2.0.0
+			 *
+			 * @param int   $activity_id Activity item id.
+			 * @param array $children    array of children item of comment.
+			 */
+			do_action( 'bp_activity_delete_comment_children', $activity_id, $child->id );
```
Comment (dipskakadiya) [inline src/bp-activity/bp-activity-functions.php:3520]: @janakkaneriya007 Above action is not needed as `bp_activity_delete_children` function use `bp_activity_delete` function. That funcation has hooks & filter.

File: src/bp-moderation/classes/class-bp-moderation-messages.php [hunk 5]
```diff
@@ -52,6 +52,10 @@ public function __construct() {
 
 		// button class.
 		add_filter( 'bp_moderation_get_report_button_args', array( $this, 'update_button_args' ), 10, 3 );
+
+		// Delete message moderation data when message thread is deleted.
+		//add_action( 'bp_messages_thread_after_delete', array( $this, 'delete_moderation_data' ), 10, 4 );
```
Comment (dipskakadiya) [inline src/bp-moderation/classes/class-bp-moderation-messages.php:57]: @janakkaneriya007 Why this action is commented. if it's not needed then remove it. Also, remove `delete_moderation_data` function if not needed

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-activity/bp-activity-functions.php (hunk 4 above)
Comment (janakkaneriya007) [reply src/bp-activity/bp-activity-functions.php:3520]: @dipskakadiya We need to keep this as it iterates with all the child comments of the activity and any of those chid comments is moderated then we must need to delete it and it's the last hook where we are getting children comments

**Review by dipskakadiya — APPROVED** [review]

**Pull request merged**